internal/job/repository_job_run.go

```diff
@@ -0,0 +1,199 @@
+	}
+	var dbMask, nullFields []string
+	dbMask, nullFields = common.BuildUpdatePaths(
+		map[string]interface{}{
+			"EndTime":        run.EndTime,
+			"CompletedCount": run.CompletedCount,
+			"TotalCount":     run.TotalCount,
+			"Status":         run.Status,
+		},
+		fieldMaskPaths,
+		nil,
+	)
+	if len(dbMask) == 0 && len(nullFields) == 0 {
+		return nil, db.NoRowsAffected, errors.New(errors.EmptyFieldMask, op, "empty field mask")
+	}
+
+	var rowsUpdated int
+	run = run.clone()
+	_, err := r.writer.DoTx(ctx, db.StdRetryCnt, db.ExpBackoff{},
+		func(_ db.Reader, w db.Writer) error {
+			var err error
+			rowsUpdated, err = w.Update(ctx, run, dbMask, nullFields)
+			if err != nil {
+				return errors.Wrap(err, op)
+			}
+			if rowsUpdated > 1 {
+				return errors.New(errors.MultipleRecords, op, "more than 1 resource would have been updated")
+			}
+			return nil
+		},
+	)
+	if err != nil {
+		return nil, db.NoRowsAffected, errors.Wrap(err, op)
+	}
+
+	return run, rowsUpdated, nil
+}
+
+// LookupJobRun will look up a run in the repository. If the run is not
+// found, it will return nil, nil.
+//
+// All options are ignored.
+func (r *Repository) LookupJobRun(ctx context.Context, id int64, _ ...Option) (*JobRun, error) {
+	const op = "job.(Repository).LookupJobRun"
+	if id == 0 {
+		return nil, errors.New(errors.InvalidParameter, op, "missing id")
+	}
+
+	run := allocJobRun()
+	if err := r.reader.LookupWhere(ctx, run, "id = ?", id); err != nil {
+		if errors.IsNotFoundError(err) {
+			return nil, nil
+		}
+		return nil, errors.Wrap(err, op, errors.WithMsg(fmt.Sprintf("failed for %d", id)))
+	}
+	return run, nil
+}
+
+// ListJobRuns returns a slice of runs for the jobId.
+// WithLimit is the only option supported.
+func (r *Repository) ListJobRuns(ctx context.Context, jobId string, opt ...Option) ([]*JobRun, error) {
+	const op = "job.(Repository).ListJobRuns"
+	if jobId == "" {
+		return nil, errors.New(errors.InvalidParameter, op, "missing job id")
+	}
+	opts := getOpts(opt...)
+	limit := r.limit
+	if opts.withLimit != 0 {
+		// non-zero signals an override of the default limit for the repo.
+		limit = opts.withLimit
+	}
+	var runs []*JobRun
+	err := r.reader.SearchWhere(ctx, &runs, "job_id = ?", []interface{}{jobId}, db.WithLimit(limit))
+	if err != nil {
+		return nil, errors.Wrap(err, op)
+	}
+	return runs, nil
+}
+
+// DeleteJobRun deletes the job for the provided id from the repository
+// returning a count of the number of records deleted.
+//
+// All options are ignored.
+func (r *Repository) DeleteJobRun(ctx context.Context, id int64, _ ...Option) (int, error) {
+	const op = "job.(Repository).DeleteJobRun"
+	if id == 0 {
+		return db.NoRowsAffected, errors.New(errors.InvalidParameter, op, "missing id")
+	}
+
+	run := allocJobRun()
+	var rowsDeleted int
+	_, err := r.writer.DoTx(
+		ctx, db.StdRetryCnt, db.ExpBackoff{},
+		func(_ db.Reader, w db.Writer) (err error) {
+			rowsDeleted, err = w.Delete(ctx, run, db.WithWhere("id = ?", id))
```

Comment:
I'm not sure this where clause is needed, if `id` is tagged as the primary key.

Reply:
ty